src/rpc/core_rpc_server.cpp

```diff
@@ -2185,6 +2196,7 @@ namespace cryptonote
     if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_BLOCK_HEADERS_RANGE>(invoke_http_mode::JON_RPC, "getblockheadersrange", req, res, r))
       return r;
 
+    db_rtxn_guard rtxn_guard(&m_core.get_blockchain_storage().get_db());
```

Comment:
Of course not terribly important, but if we are at it anyway already, the test about restriction could come first, then the lock, and only then the check against current blockchain height.